### PR TITLE
Jshint improvements

### PR DESCRIPTION
**Pull request by Graham42**
This continues the work started by @baer in https://github.com/adamwdraper/Numeral-js/pull/100. I adjusted the original proposal to be a little less strict because we need things like bitwise operators.

What's here now seems like a good starting point so I'm opening this PR for a place of discussion for how strict jshint rules should be.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  // == Enforcing Options ===============================================
+  //
+  // These options tell JSHint to be more strict towards your code.
+
+  "bitwise"       : false,     // Prohibit bitwise operators (&, |, ^, etc.).
+  "camelcase"     : true,     // Force all variable names to use either camelCase style or UPPER_CASE with underscores.
+  "curly"         : true,     // Require {} for every new block or scope.
+  "eqeqeq"        : true,     // Require triple equals i.e. `===`.
+  "es3"           : false,    // Adhere to ECMAScript 3 specification for legacy browsers like Internet Explorer 6/7/8.
+  "forin"         : true,     // Tolerate `for in` loops without `hasOwnPrototype`.
```
Comment (BenjaminVanRyseghem) [inline .jshintrc]: I think that should be `false` and `forEach` should be used instead

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  "immed"         : true,     // Require immediate invocations to be wrapped in parens e.g. `( function(){}() );`
+  "indent"        : 4,        // Specify indentation spacing
+  "latedef"       : true,     // Prohibit variable use before definition.
+  "newcap"        : true,     // Require capitalization of all constructor functions e.g. `new F()`.
+  "noarg"         : true,     // Prohibit use of `arguments.caller` and `arguments.callee`.
+  "noempty"       : true,     // Prohibit use of empty blocks.
+  "nonew"         : true,     // Prohibit use of constructors for side-effects.
+  "plusplus"      : false,    // Prohibit use of `++` & `--`.
+  "quotmark"      : "single", // Forces single quotes for strings
```
Comment (BenjaminVanRyseghem) [inline .jshintrc]: Isn't `"` the standard?

File: .jshintrc [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  // == Relaxing Options ================================================
+  //
+  // These options allow you to suppress certain types of warnings. Use
+  // them only if you are absolutely positive that you know what you are
+  // doing.
+
+  "asi"           : false,    // Tolerate Automatic Semicolon Insertion (no semicolons).
+  "boss"          : false,    // Tolerate assignments inside if, for & while. Usually conditions & loops are for comparison, not assignments.
+  "debug"         : true,     // Allow debugger statements e.g. browser breakpoints.
```
Comment (BenjaminVanRyseghem) [inline .jshintrc]: I think this one should be `false` :smile:

File: .jshintrc (hunk 2 above)
Comment (Graham42) [inline .jshintrc]: `"` is required for JSON, but javascript doesn't actually care

File: .jshintrc (hunk 1 above)
Comment (Graham42) [inline .jshintrc]: The meaning of this is not which loop to use, but rather to enforce checking inside a `for..in` loop which is meant to make you check `hasOwnProperty`  on keys of an object.

Also, `forEach` is actually not as fast as a regular for loop http://jsperf.com/for-vs-foreach/313.

**Comment by BenjaminVanRyseghem** [comment]
:+1:

**Comment by BenjaminVanRyseghem** [comment]
In numbro 1.2.0

**Pull request merged**